### PR TITLE
Enable Python 3.5

### PR DESCRIPTION
**Pull request by QuLogic**
Not sure if I changed everything necessary here, but I guess we'll see. I tested all the builds though, so they should be okay.

### CODE REVIEW

**Comment by QuLogic** [comment]
So half of these are already built for some reason, but no matter. Is this a bug in ObviousCI/conda-build, or should I be doing something differently:
```
Building  cartopy-0.13.0-np18py34_2
Removing old build environment
Removing old work directory
BUILD START: cartopy-0.13.0-np18py34_2
Fetching package metadata: ........
Solving package specifications: ...............
The following NEW packages will be INSTALLED:
    cython:          0.23.2-py35_0   defaults                                 
    and a bunch of Python 3.5 packages.
```

**Comment by QuLogic** [comment]
Also, from circleci:
```
Proceed ([y]/n)? Invalid choice: conda info
Proceed ([y]/n)? Invalid choice: unset lang
Proceed ([y]/n)? Invalid choice: yum install -y expat-devel git autoconf libtool texinfo check-devel
Proceed ([y]/n)? 
```

It should be `conda install -y anaconda-client`, I guess?

**Comment by QuLogic** [comment]
And on AppVeyor, Cartopy for 2.7 actually built against 3.4?

**Comment by QuLogic** [comment]
Even with the new ObviousCI, something is not quite right. Cartopy is not supposed to build on 3.5, but it still is. Then when it builds Cartopy for 2.7, it builds on 3.5, then tests on 2.7.

File: .travis.yml [hunk 1]
```diff
@@ -37,5 +37,7 @@ install:
     - obvci_install_conda_build_tools.py
 
 script:
-    - obvci_conda_build_dir.py ./ SciTools --channel main --build-condition "numpy >=1.8" "python >=2.7,<3|>=3.4,<3.5"
```
Comment (pelson) [inline .travis.yml]: This breakdown shouldn't be necessary (I appreciate it is probably just an attempt to separate the concerns so that it can be tested)

**Comment by pelson** [comment]
Is this a case of conda going mad? Could you try adding ```python x.x``` as the run (maybe even build) dependency? If the latter is needed, you will have to use the ``skip`` mechanism to control whether we build against 35 or not for cartopy until conda-build is fixed to have sensible behaviour.

File: .travis.yml (hunk 1 above)
Comment (QuLogic) [inline .travis.yml]: Yes, it did seem to fix CircleCI, but not Travis for whatever reason.

**Comment by QuLogic** [comment]
Cartopy already depends on `python >=2.7,<3.5`; what does `python x.x` do?

**Comment by pelson** [comment]
> Cartopy already depends on python >=2.7,<3.5; what does python x.x do?

I'm less and less sure the more I'm using the ``x.x`` stuff. I'm trying to sort out how it has been implemented in https://github.com/conda/conda-build/pull/650.

**Comment by QuLogic** [comment]
OK, I'm not sure that would help then. The problem is clearly in the build step. There seems to be an issue with both ObviousCI and conda.
 * ObviousCI attempts to build Cartopy against 3.5, even though it's against the build requirements. (We should probably just work on getting Cartopy working with 3.5, though.)
 * conda misbuilds things like `cartopy-0.13.0-np110py27_2` even though `CONDA_PY=27`, requirements say <3.5, it's still building against 3.5! Then it tries to test on 2.7, which makes no sense.

File: cartopy/meta.yaml [hunk 2]
```diff
@@ -10,12 +10,11 @@ source:
...
 
 build:
-  number: 0  # [not linux]
-  number: 1  # [linux]
+  number: 2
 
 requirements:
   build:
-    - python
+    - python >=2.7,<3.5  # See https://github.com/SciTools/cartopy/pull/676
```
Comment (pelson) [inline cartopy/meta.yaml]: Since we're having problems, try specifying this in the skip section:

```
build:
   skip: True  # [py3k]
```

File: scripts/run_docker_build.sh [hunk 3]
```diff
@@ -44,6 +44,8 @@ conda info
 unset LANG
 yum install -y expat-devel git autoconf libtool texinfo check-devel
 
-obvci_conda_build_dir /conda-recipes $UPLOAD_OWNER --build-condition "numpy >=1.8" "python >=2.7,<3|>=3.4,<3.5"
```
Comment (pelson) [inline scripts/run_docker_build.sh]: I think we can just revert this change, and remove the ``<3.5``.
